### ISSUE TITLE
High memory usage when ingesting scrypt-based keystores

### ISSUE THREAD

**Issue by benjaminion** [Consensys-Incorporated/teku#2492]
Tl;dr: implement a flag to limit the parallelism when decrypting keystores (edit: better idea below).

When loading scrypt-based keystores there is a trade-off in time and memory: scrypt is both slow and memory-intensive.

PR #2437 improved keystore loading times by decrypting keystores in parallel. However, this can require large amounts of initial heap space - much more than we need for normal running. It would be good to be able to start with lower heap requirements: maybe a flag to set the max parallelism.

An experiment on my test system. I inserted an exit after loading validators in the `initializeValidators()` method and tested how much heap was needed to load the full set without an out of heap error. Heap size set by `-XmxNNNNm`.

| # keystores | MB heap required |
| ---| --- |
| 0 | 15 |
| 1 | 400 |
| 2 | 600 |
| 3 | 900 |
| 4 | 1200 |
| 5 | 1500 |
| 6 | 1800 |
| 7 | 2100 |
| 8 | 2400 |
| >8 | 2400 |

The heap requirement tops out at 2400mb, which I assume indicates that the JVM is using 8-way parallelism on this system (it is an 8 processor linux VM).

Note that this does _not_ occur with pbkdf2 based keystores. I can load 64 of those in 19mb of heap. (It is also instant.)

**Comment by benjaminion** [comment]
Rather than increasing the number of config options and the resulting documentation, maintenance, and support headache...

If Java can see its own max heap size, we can calculate the optimal parallelism given the available memory. Basically this many threads: `(Xmx - margin) / sizeof(scrypt data)`, where `margin` is either the current heap usage (after triggering a GC?) or some known amount of overhead memory we need at this point.
